src/main/groovy/de/dkfz/roddy/execution/jobs/SubmissionCommand.groovy

```diff
@@ -1,154 +1,159 @@
+
+    /**
+     * The command which should be called
+     */
+    protected String command
+
+    protected List<String> dependencyIDs
+
+    protected final List<ProcessingParameters> processingParameters
+    /**
+     * A command to be executed on the cluster head node, in particular qsub, bsub, qstat, etc.
+     *
+     * @param parentJobManager
+     * @param job
+     * @param jobName
+     * @param environmentVariables
+     *
+     */
+    protected SubmissionCommand(BatchEuphoriaJobManager parentJobManager, BEJob job, String jobName,
+                                List<ProcessingParameters> processingParameters,
+                                Map<String, String> environmentVariables, List<String> dependencyIDs,
+                                String command) {
+        super(parentJobManager, job, jobName, environmentVariables)
+        this.processingParameters = processingParameters
+        this.command = command
+        this.dependencyIDs = dependencyIDs ?: new LinkedList<String>()
+    }
+
+    /**
+     * Should the local environment be passed?
+     *
+     * JobManager and SubmissionCommand together determine, whether the environment should be passed.
+     *
+     * * The Command has precedence over the JobManager value.
+     * * If the Command value is not set, the JobManager value is the fallback.
+     * * If neither JobManager nor Command are defined, only copy the requested variables to remote.
+     *
+     * @return
+     */
+    Boolean getPassLocalEnvironment() {
+        passEnvironment.orElse(parentJobManager.passEnvironment)
+    }
+
+    @Override
+    String toBashCommandString() {
+        String email = parentJobManager.getUserEmail()
+        String umask = parentJobManager.getUserMask()
+        String groupList = parentJobManager.getUserGroup()
+        boolean holdJobsOnStart = parentJobManager.isHoldJobsEnabled()
+
+        // collect parameters for job submission
+        List<String> parameters = []
+        parameters << assembleVariableExportParameters()
+        parameters << getAccountNameParameter()
+        parameters << getJobNameParameter()
+        if (holdJobsOnStart) parameters << getHoldParameter()
+        parameters << getWorkingDirectoryParameter()
+        parameters << getLoggingParameter(job.jobLog)
+        parameters << getEmailParameter(email)
+        if (groupList && groupList != "UNDEFINED") parameters << getGroupListParameter(groupList)
+        parameters << getUmaskString(umask)
+        parameters << assembleProcessingCommands()
+        parameters << assembleDependencyParameter(creatingJob.parentJobIDs)
+        parameters << getAdditionalCommandParameters()
+
+        // create job submission command call
+        StringBuilder command = new StringBuilder(EMPTY)
+
+        if (environmentString) {
+            command << "${environmentString} "
+        }
+
+        if (job.toolScript) {
+            if (parentJobManager instanceof SlurmJobManager){

```

Comment:
`instanceof` is almost always an indication of a design problem.

Furthermore, did you consider adding the shebang line independent of the cluster type? In principle, it really shouldn't make a difference, whether the script has a shebang line, if it works with no shebang -- unless the toolScript is not Bash code, of course.